File: packages/core/src/prompt/PromptModels.ts
/**
 * A mitigation prompt entry
 */
export interface MitigationPrompt {
  id: string;
  description: string;
}

/**
 * A rule hint from heuristic matching
 */
export interface RuleHint {
  signalId: string;
  line: number;
  snippet: string;
}

/**
 * A code chunk for analysis
 */
export interface CodeChunkPrompt {
  id: string;
  file: string;
  startLine: number;
  endLine: number;
  code: string;
  ruleHints: RuleHint[];
}

/**
 * A technique prompt for batch mode
 */
export interface TechniquePrompt {
  id: string;
  name: string;
  severity: string;
  summary: string;
  description: string;
  mitigations: MitigationPrompt[];
  ruleHints: RuleHint[];
}

/**
 * Single technique prompt payload
 */
export interface PromptPayload {
  techniqueId: string;
  techniqueName: string;
  severity: string;
  summary: string;
  description: string;
  mitigations: MitigationPrompt[];
  codeChunk: CodeChunkPrompt;
  readmeExcerpt?: string;
}

/**
 * Batch prompt payload for multiple techniques
 */
export interface BatchPromptPayload {
  techniques: TechniquePrompt[];
  codeChunk: CodeChunkPrompt;
  readmeExcerpt?: string;
}

File: packages/intellij-plugin/src/main/kotlin/ai/astha/scantheplanet/idea/scanner/prompt/PromptModels.kt
package ai.astha.scantheplanet.idea.scanner.prompt

import ai.astha.scantheplanet.idea.scanner.RuleHint

data class PromptPayload(
    val techniqueId: String,
    val techniqueName: String,
    val severity: String,
    val summary: String,
    val description: String,
    val mitigations: List<MitigationPrompt>,
    val codeChunk: CodeChunkPrompt,
    val readmeExcerpt: String?
)

data class BatchPromptPayload(
    val techniques: List<TechniquePrompt>,
    val codeChunk: CodeChunkPrompt,
    val readmeExcerpt: String?
)

data class TechniquePrompt(
    val id: String,
    val name: String,
    val severity: String,
    val summary: String,
    val description: String,
    val mitigations: List<MitigationPrompt>,
    val ruleHints: List<ai.astha.scantheplanet.idea.scanner.RuleHint>
)

data class MitigationPrompt(
    val id: String,
    val description: String
)

data class CodeChunkPrompt(
    val id: String,
    val file: String,
    val startLine: Int,
    val endLine: Int,
    val code: String,
    val ruleHints: List<RuleHint>
)
